src/protocol/server.c

```diff
@@ -198,7 +253,166 @@ protocolServerProcess(
+                                                if (session->id == this->sessionId)
+                                                {
+                                                    sessionData = session->data;
+                                                    break;
+                                                }
+                                            }
+
+                                            // Error when session not found
+                                            if (sessionData == NULL && request.type != protocolCommandTypeCancel)
+                                            {
+                                                THROW_FMT(
+                                                    ProtocolError, "unable to find session id %" PRIu64 " for request %s:%s",
+                                                    this->sessionId, strZ(strIdToStr(request.id)),
+                                                    strZ(strIdToStr(request.type)));
+                                            }
+                                        }
+
+                                        // Process request type
+                                        switch (request.type)
+                                        {
+                                            // Process request
+                                            case protocolCommandTypeProcess:
+                                            {
+                                                // Process session
+                                                ProtocolServerResult *processResult;
+
+                                                if (handler->processSession != NULL)
+                                                {
+                                                    ASSERT(handler->process == NULL);
+                                                    CHECK_FMT(
+                                                        ProtocolError, this->sessionId != 0, "no session id for request %s:%s",
+                                                        strZ(strIdToStr(request.id)), strZ(strIdToStr(request.type)));
+
+                                                    processResult = handler->processSession(pckReadNew(request.param), sessionData);
+                                                }
+                                                // Standalone process
+                                                else
+                                                {
+                                                    ASSERT(handler->process != NULL);
+                                                    ASSERT(!request.sessionRequired);
+
+                                                    processResult = handler->process(pckReadNew(request.param));
+                                                }
+
+                                                if (processResult == NULL)
+                                                    protocolServerResponseP(this);
```

Comment:
So a `NULL` result is not an error here; what are the types of handlers' processes that return `NULL` here?

Reply:
Handlers that do not have any data to return return NULL. Something like storageRemotePathSyncProtocol() is a good example. It will either succeed or throw an exception, which is handled separately.